workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/GameObjectCreation/SpatialOSEntity.cs

```diff
@@ -0,0 +1,32 @@
+﻿using Improbable.Worker;
+using Unity.Entities;
+
+namespace Improbable.Gdk.Core.GameObjectRepresentation
+{
+    /// <summary>
+    ///     Used to easily retrieve information about a SpatialOS Entity instance from a Unity ECS Entity instance.
+    /// </summary>
+    public struct SpatialOSEntity
+    {
+        public readonly EntityId SpatialOSEntityId;
+        private readonly EntityManager entityManager;
+        private readonly Entity entity;
+
+        internal SpatialOSEntity(Entity entity, EntityManager entityManager)
+        {
+            this.entity = entity;
+            this.entityManager = entityManager;
+            SpatialOSEntityId = entityManager.GetComponentData<SpatialEntityId>(entity).EntityId;
+        }
+
+        public bool HasComponent<T>() where T : struct, ISpatialComponentData, IComponentData
```

Comment:
`ISpatialComponentData` already indicates `IComponentData`, no need to specify both

Reply:
is this on recent master?

Reply:
ISpatialComponentData could/should extend `IComponentData` for this to work

Reply:
I guess `HasComponent` may not have the restriction, if that's the case, 👍

Reply:
On current master the two always coincide but ISpatialComponentData doesn't extend IComponentData. So we unfortunately need both constants :(.